packages/core/src/VRange.ts

```diff
@@ -233,6 +233,51 @@ export class VRange {
         }
         return this;
     }
+    /**
+     * Split the range containers up to their common ancestor. Return all
+     * children of the common ancestor that are targeted by the range after the
+     * split. If a predicate is given, splitting continues up to and including
+     * the node closest to the common ancestor that matches the predicate.
+     *
+     * @param predicate
+     */
+    split<T>(predicate?: Predicate<T>): VNode[] {
+        const ancestor = this.startContainer.commonAncestor(this.endContainer);
+        const closest = ancestor.closest(predicate);
+        const container = closest ? closest.parent : ancestor;
+
+        // Split the start ancestors.
+        let start = this.start;
+        do {
+            let startAncestor = start.parent;
+            if (startAncestor.children.indexOf(start) > 0) {
+                startAncestor = startAncestor.splitAt(start);
+            }
+            start = startAncestor;
+        } while (start.parent !== container);
+
+        // Split the end ancestors.
+        let end = this.end;
+        do {
+            const endAncestor = end.parent;
+            const children = endAncestor.children;
+            if (children.indexOf(end) < children.length - 1) {
+                endAncestor.splitAt(end);
+                endAncestor.append(end);
+            }
+            end = endAncestor;
+        } while (end.parent !== container);
+
+        // Return all top-most split nodes between and including start and end.
+        const nodes = [];
+        let node = start;
+        while (node !== end) {
+            nodes.push(node);
+            node = node.nextSibling();
```

Comment:
Shouldn't it be `next`? What if the nodes have different parents?
_Brotha from an otha motha_ :call_me_hand:

Reply:
I don't think so because we return only the top-most containers that were split, so they are all consecutive siblings of the container we did not split, which is eithier their common ancestor or an ancestor of their common ancestor.